engine/src/main/java/org/terasology/engine/core/module/ModuleManager.java

```diff
@@ -105,49 +98,61 @@ public ModuleManager(String masterServerAddress, List<Class<?>> classesOnClasspa
 
     private void loadModulesFromApplicationPath(PathManager pathManager) {
-        ModulePathScanner scanner = new ModulePathScanner(new ModuleLoader(metadataReader));
-        scanner.getModuleLoader().setModuleInfoPath(TerasologyConstants.MODULE_INFO_FILENAME);
-        scanner.scan(registry, pathManager.getModulePaths());
+        ModulePathScanner scanner = new ModulePathScanner(moduleFactory);
+        List<File> paths = pathManager.getModulePaths().stream().map(Path::toFile).collect(Collectors.toList());
+        scanner.scan(registry, paths);
     }
 
     private Module loadEngineModule(List<Class<?>> classesOnClasspathsToAddToEngine) {
-        Module engineModule;
-        try (Reader reader = new InputStreamReader(getClass().getResourceAsStream("/engine-module.txt"), TerasologyConstants.CHARSET)) {
-            ModuleMetadata metadata = metadataReader.read(reader);
-            List<Class<?>> additionalClassesList = new ArrayList<>(classesOnClasspathsToAddToEngine.size() + 2);
-            additionalClassesList.add(Module.class); // provide access to gestalt-module.jar
-            additionalClassesList.add(Asset.class); // provide access to gestalt-asset-core.jar
-            additionalClassesList.add(UIWidget.class); // provide access to nui.jar
-            additionalClassesList.add(TypeRegistry.class); // provide access to nui-reflect.jar
-            additionalClassesList.add(KeyboardDevice.class); // provide access to nui-input.jar
-            additionalClassesList.add(ModuleTypeRegistry.class); // provide access to nui-gestalt7.jar
-            additionalClassesList.addAll(classesOnClasspathsToAddToEngine); // provide access to any facade-provided classes
-            Class<?>[] additionalClassesArray = new Class[additionalClassesList.size()];
-            additionalClassesArray = additionalClassesList.toArray(additionalClassesArray);
-            engineModule = ClasspathModule.create(metadata, getClass(), additionalClassesArray);
-        } catch (IOException e) {
-            throw new RuntimeException("Failed to read engine metadata", e);
-        } catch (URISyntaxException e) {
-            throw new RuntimeException("Failed to convert engine library location to path", e);
-        }
+        Module packageModule = moduleFactory.createPackageModule("org.terasology.engine");
+
+        // We need to add reflections from our subsystems and other classes.
+        ConfigurationBuilder config = new ConfigurationBuilder();
+        Reflections packageReflections = packageModule.getModuleManifest();
+        Set<Scanner> scanners = packageReflections.getConfiguration().getScanners();
+        config.setScanners(scanners.toArray(new Scanner[0]));
 
-        enrichReflectionsWithSubsystems(engineModule);
+        for (Class<?> aClass : classesOnClasspathsToAddToEngine) {
+            config.addUrls(ClasspathHelper.forClass(aClass));
+        }
 
-        return engineModule;
+        // TODO: is this using reflections.cache?
```

Comment:
How can we figure this out?

Reply:
Oh, I think I did figure that one out: When using ModuleFactory's createDirectoryModule or createArchiveModule, then it will check for a reflections.cache to load. But there's no implicit cache checking when using the Reflections API directly, it only reads from a cache if you call those methods.

Reply:
How often do we call these methods? Can we do some crude benchmarking for an Omega workspace and run it on a couple of developer machines to get a feeling when the reflection cache helps, and how big the impact of removing it is?

Does this only affect local development, or also our distributions?

Reply:
Not called too often. I think it's just once per GameEngine.initialize, which looks like it's once per process.

(not counting tests, which want to make new engines or engine-like things a whole bunch.)

> Does this only affect local development, or also our distributions?

all.